packages/calypso-e2e/src/lib/pages/login-page.ts

```diff
@@ -43,75 +54,113 @@ export class LoginPage {
 	}
 
 	/**
 	 * Switch account if an user is already logged in.
 	 */
 	private async switchAccount(): Promise< void > {
-		// By default, log out of the existing account (even if test steps end up logging back in).
-		const alreadyLoggedIn = await this.page.$( selectors.changeAccountButton );
+		await this.pageSettled();
+
+		// Change account button takes some time to fade into view if an account
+		// is already logged in.
+		const alreadyLoggedIn = await this.page.isVisible( selectors.continueAsUser );
 		if ( alreadyLoggedIn ) {
-			console.log( 'already logged in, selecting "change account' );
+			const elementHandle = await this.page.waitForSelector( selectors.continueAsUser );
+			await elementHandle.waitForElementState( 'stable' );
 			await this.page.click( selectors.changeAccountButton );
 		}
 	}
 
 	/**
-	 * Executes series of interactions on the log-in page to log in as a specific user.
+	 * The basic flow of filling out username/email and password into the form then submitting.
 	 *
-	 * @param {Object} param0 Key/value pair holding the credentials for a user.
-	 * @param {string} param0.username Username of the user.
-	 * @param {string} param0.password Password of the user.
-	 * @returns {Promise<void>} No return value.
+	 * Call this method from any page to log into WordPress.com.
+	 *
+	 * @param param0 LoginCredentials object.
+	 * @param {string} param0.username User name.
+	 * @param {string} param0.password Password.
 	 */
-	async login( { username, password }: { username: string; password: string } ): Promise< void > {
-		await this.pageSettled();
-
-		await this.switchAccount();
-
-		// Begin the process of logging in.
+	private async baseflow( { username, password }: LoginCredentials ): Promise< void > {
 		await this.page.fill( selectors.username, username );
 		await this.page.keyboard.press( 'Enter' );
 		await this.page.fill( selectors.password, password );
-		await this.page.click( selectors.loginButton );
-		// Make sure after logging in that everything stablizes, so we can continue with the next action!
+		// Wait for navigation to resolve in a regular login.
+		await Promise.all( [
+			this.page.waitForNavigation(),
+			this.page.click( selectors.loginButton ),
+		] );
+		// After navigation resolves, wait for `load` state to fire.
 		await this.page.waitForLoadState( 'load' );
 	}
 
+	/* Log in methods */
+
 	/**
-	 * Navigates to the Login page.
+	 * Log in to WordPress.com from the `/log-in` endpoint.
+	 *
+	 * This is the 'normal' or 'standard' way of performing log ins.
+	 *
+	 * @param {LoginCredentials | TestAccount} credentials Credentials of the user. Specify either an username/password pair or name of a test account in the configuration.
 	 */
-	async visit(): Promise< void > {
-		await this.page.goto( getCalypsoURL( 'log-in' ), { waitUntil: 'networkidle' } );
+	async login( credentials: LoginCredentials | TestAccount ): Promise< void > {
```

Comment:
This is the method that most test writers will interact with, and it allows the test writer to specify two types of credentials.

I kept this behavior from existing `LoginFlow` as it would be quite tedious to retrieve the LoginCredentials pair for test users stored in the configuration within the spec file. I think the convenience of allowing users to specify the test user outweighs any complexity.

One note - whereas `LoginFlow` had the default user of `defaultUser` if a test account was not specified, the refactored code will force test writers to explicitly specify the test account.